frontend/src/container/Home.jsx

```diff
@@ -65,7 +66,7 @@ export class Home extends React.Component {
   }
 
   componentDidMount () {
-    this.checkUsername()
+    this.maybeShowUsernamePopup()
```

Comment:
I don't love that a function starts with maybe, some other possibilities: `setDisplayUsernamePopup` or `setUsernamePopupVisibility`
This comment is just a suggestion, if you prefer your name yet, you can just resolve the conversation.